media-proxy/src/rdma_session.c

```diff
@@ -345,6 +345,11 @@ static void *rx_rdma_frame_thread(void *arg)
     libfabric_ctx *rdma_ctx = s_ctx->rdma_ctx;
     ep_ctx_t *cp_ctx = s_ctx->ep_ctx;
 
+    while (!atomic_load(&s_ctx->shm_ready) && !s_ctx->stop) {
+        INFO("%s memif not ready\n", __func__);
+        usleep(1000);
+    }
+
     printf("%s(%d), start\n", __func__, s_ctx->idx);
     while (!s_ctx->stop) {
         ep_recv_buf(cp_ctx, cp_ctx->data_buf, s_ctx->transfer_size);
```

Comment:
```suggestion
        ep_recv_buf(cp_ctx, cp_ctx->data_buf, s_ctx->transfer_size);
        if (!atomic_load_explicit(&s_ctx->shm_ready, memory_order_relaxed)
            continue;
```

Reply:
I used `memory_order_acquire` instead of `memory_order_relaxed` because relaxed is ok when the state changes to not ready and we want to skip rx_rdma_consume_frame(), but then if shmem is again ready we need `memory_order_acquire`